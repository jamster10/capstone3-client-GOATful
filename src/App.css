.page{
    animation-name: test;
    animation-duration: .5s;
}
::-webkit-scrollbar { 
    display: none; 
}
@keyframes test {
    0% {
        opacity: 0;
    }
    100%{
        opacity: 1;
    }
}

.footer a{
    color: white;
}

footer {
  background-color: rgba(0, 0, 0, 0.66);
  width: 100%;
  height: 60px;
  color: white;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

footer > span {
  padding: .75em;
}