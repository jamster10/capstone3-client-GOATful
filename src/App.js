import React , {Component} from 'react';
import { Route, Switch } from 'react-router-dom'
import logo from './logo.svg';
import PrivateRoute from './protected/PrivateRoute'
import PublicOnlyRoute from './protected/PublicOnlyRoute'

import Landing from './routes/Landing'
import AddVid from './routes/AddVideo'
import Creator from './routes/Creator'
import Login from './routes/Login'
import VidPrev from './routes/VidPreview'
import Videos from './routes/Videos'
import NotFound from './routes/Oops'
import './App.css';


export default class App extends Component {
  state = {
    hasError : null,
  }

  render(){

    const landPage = () => {
      return(
        <Landing />
      )
    }

    const loginPage = () => {
      return(
        <Login />
      )
    }

    const creatorPage = () => {
      return(
        <Creator />
      )
    }

    const addVidPage = () => {
      return(
        <AddVid />
      )
    }

    const vidPrevPage = () => {
      return(
        <VidPrev />
      )
    }

    const videosPage = () => {
      return(
        <Videos />
      )
    }

    return(
      <div className="App">
        <div>
          this is the navBar
        </div>
        <main>
          <Switch>
            <PublicOnlyRoute
              exact
              path={'/'}
              component={landPage}
            />
            <Route
              exact
              path={'/creator'}
              component={creatorPage}
            />
            <PublicOnlyRoute
              exact
              path={'/add-video'}
              component={addVidPage}
            />
            <PublicOnlyRoute
              exact
              path={'/login'}
              component={loginPage}
            />
            <PrivateRoute 
              exact
              path={'/'}
              component={videosPage}
            />
            <PrivateRoute
              exact
              path={'/:vid-id/previews'}
              component={vidPrevPage}
            />
            <Route
              component={NotFound}
            />
          </Switch>
        </main>
      </div>
    )
  }
}

