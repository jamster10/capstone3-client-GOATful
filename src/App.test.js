// import React from 'react';
// import ReactDOM from 'react-dom';
// import renderer from 'react-test-renderer'
// import { configure } from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';

// import App from './App';

// import { shallow } from 'enzyme'
// import toJson from 'enzyme-to-json'

// configure({ adapter: new Adapter() });

// it('NavBar renders without crashing', () => {
//     const wrapper = shallow(<App />)
//     expect(toJson(wrapper)).toMatchSnapshot()
// });


import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';

describe('App Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(< App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
})
