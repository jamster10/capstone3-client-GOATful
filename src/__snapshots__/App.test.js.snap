// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders without crashing 1`] = `
<div
  className="App"
>
  <NavBar />
  <main>
    <Switch>
      <PublicOnlyRoute
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/creator"
      />
      <PublicOnlyRoute
        component={[Function]}
        exact={true}
        path="/add-video"
      />
      <PublicOnlyRoute
        component={[Function]}
        exact={true}
        path="/login"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/:vid-id/previews"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </main>
</div>
`;
